### PR TITLE
US024 - Registration System

### PR DESCRIPTION
**Pull request by BlankAmber**
# Pull Request Checklist

Please follow all these rules before merging your branch.
- [x] Must reference a ticket
- [x] Must implement tests where appropirate
- [x] Must have useful documentation
- [x] Must adhere to the naming conventions
- [x] Must not commit confidential information 

Once all requirements above are fulfilled, your pull request has no merge conflicts, and you have two approvals from the approved approvers, then it can be tested and signed off.

# PR dependencies
- #18 

# Relevant tickets
- Implements backend for [US024 - Registration System](https://tasks.office.com/kainos.com/Home/Task/-LpElUzK40epI97zNycNUZYABeGF?Type=TaskLink&Channel=Link&CreatedTime=638385025729370000)

# Changes
- Adds a POST /register endpoint
  - Only admins can register accounts
  - Allows you to register either employee accounts or admin accounts
- Fixes a bug with `DaoUtil.executeStatement(Connection conn, String statement, boolean isQuery, Object... values)` where non-query statements would not be executed
- Fixes a typo in `AuthIT`
- Adds JWT_SECRET to the README

# Screenshots
Screenshot 2023-12-18 at 13 33 03

# Future Considerations
- Better test coverage
  - Integration tests do not actually test registering an account (this would be complicated as the account would need to be deleted after for the integration test to be repeatable)

### CODE REVIEW

**Review by adammarshall745 — APPROVED** [review]
Looks good Samuel! This will be a very impressive addition! 👍

**Comment by BlankAmber** [comment]
TODO: Role ID when registering (only for admins)

**Comment by lullabelle** [comment]
Question - Only admins can register accounts

Where is this in the acceptance criteria?

File: src/main/java/org/kainos/ea/client/FailedToRegisterException.java [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package org.kainos.ea.client;
+
+public class FailedToRegisterException extends Exception {
+}
```
Comment (jkelso662) [inline src/main/java/org/kainos/ea/client/FailedToRegisterException.java:4]: Think someone forgot to implement this!

Comment (BlankAmber) [reply src/main/java/org/kainos/ea/client/FailedToRegisterException.java]: Oops

**Comment by BlankAmber** [comment]
> Question - Only admins can register accounts
> 
> Where is this in the acceptance criteria?

For future reference: this was addressed with the customer directly. Although our interpretation doesn't match what the customer had in mind, the customer prefers our interpretation (thank you)

**Review by jkelso662 — APPROVED** [review]
Fab1

**Pull request merged**